### PR TITLE
Migrate from Travis to GitLab CI/CD

### PR DESCRIPTION
**Pull request by JAlvarezJarreta**
_Note:_ I have already done a mirroring of this repository in GitLab.

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+# limitations under the License.
+
+# GitLab CI/CD for Python
+
+workflow:
+  rules:
+    - if: $CI_EXTERNAL_PULL_REQUEST_IID &&
+          $CI_EXTERNAL_PULL_REQUEST_SOURCE_REPOSITORY == $CI_EXTERNAL_PULL_REQUEST_TARGET_REPOSITORY &&
+          ( $CI_EXTERNAL_PULL_REQUEST_TARGET_BRANCH_NAME == $CI_DEFAULT_BRANCH ||
+            $CI_EXTERNAL_PULL_REQUEST_TARGET_BRANCH_NAME =~ /hackathon\/.+/ )
```
Comment (Dishalodha) [inline .gitlab-ci.yml:23]: I dont think we need to apply this condition here?

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+  extends: .python
+  artifacts:
+    paths:
+      - $RUN_DIR
+  before_script:
+    - python --version  # for debugging
+    - python -m venv $RUN_DIR/venv
+    - source $RUN_DIR/venv/bin/activate
+  script:
+    - pip install -e .[cicd]
```
Comment (Dishalodha) [inline .gitlab-ci.yml:62]: We might want to add the cicd dependencies in pyproject.toml file

File: .gitlab-ci.yml (hunk 1 above)
Comment (Dishalodha) [inline .gitlab-ci.yml:23]:
```suggestion
          ( $CI_EXTERNAL_PULL_REQUEST_TARGET_BRANCH_NAME == $CI_DEFAULT_BRANCH
```

Comment (JAlvarezJarreta) [reply .gitlab-ci.yml:23]: Good catch!

File: .gitlab-ci.yml (hunk 2 above)
Comment (JAlvarezJarreta) [reply .gitlab-ci.yml:61]: I completely forgot about that :facepalm:

**Review by Dishalodha — APPROVED** [review]
Looks good to me!

**Pull request merged**